### ISSUE TITLE
Add keyboard support

### ISSUE THREAD

**Issue by christophe-williams** [codingthat/old-fashioned-remote#1]
Users of smart devices often dislike entering text using on-screen keyboards - whether this to enter login credentials or to search for a title. 

Ticket is to add support for using phone keyboard to send text directly to application.

**Comment by christophe-williams** [comment]
If key mappings are independent of applications, could this feature be extended to an arbitrary application? For instance I ask xdotool to search for "Hulu" windows and use the keyboard to type login credentials, even though Hulu is not supported for other remote control features.

**Comment by codingthat** [comment]
Indeed, arbitrary keyboard input could be added (implied by the "advanced lounging" mentioned at the end of the article. ;) )  That's a bit of an extended feature though—someone would have to think about how it'd be best for the UI to work.

And yes, you can have it search for whatever you like in the title.  It would be a matter of editing `common/preset_commands.js` and adding a `Hulu` key and corresponding object, just like the others found there.  If you get it working to your liking feel free to submit a patch so other Hulu users can enjoy the results, too.
